lib/floe/workflow/states/task.rb

```diff
@@ -61,7 +61,8 @@ def finish(context)
         end
 
         def running?(context)
-          return true if waiting?(context)
+          return true  if waiting?(context)
+          return false if finished?(context)
```

Comment:
This is the fix for this issue, I'd also like to make sure we don't check running? for a sub-workflow that we know to be finished but this is also good insurance at the Task level.